pkg/devfile/parser/data/v2/2.0.0/types.go

```diff
@@ -0,0 +1,5 @@
+package version200
+
+// Devfile200 is the struct for devfile 2.0.0 specific implementation
+type Devfile200 struct {
```

Comment:
do we need this now? if not it can be removed

Reply:
We intentionally add this here for our future devfile 2.1 support. @maysunfaisal

Reply:
why we need our own structure in library, i think it will complicate things @maysunfaisal anything you have in mind that i am not able to understand.

Reply:
For minor version change of devfile schema version, they should be backward compatible, but might add additional fields in higher version. (in this case, let's say 2.1.0). They should share the same common methods defined in `/common` and `/v2`. but for 2.1.0, it might need specific functions, and might require different implementation on same function for 2.0 & 2.1. 
we define our own struct like that in order to support both 2.0 and 2.1. 
This is an example branch we used for testing, which 2.0 and 2.1 have different implementation on `GetCustomType()`: https://github.com/maysunfaisal/parser/commit/3fd444c23f259b55da551760efb4c5633585cda3

Reply:
For now, the empty struct looks like meaning less; we just set it up for future 2.1 support.

Reply:
@adisky Yes, the idea behind this is; there may be feature differences between 2.0.0 & 2.1.0. For example, 2.1.0 will be having build guidance which 2.0.0 is not going to support. Hence we laid out the dir 2.0.0 & 2.1.0 with their own structs and an  interface pkg/devfile/parser/data/v2/interface.go(empty for now) which will have the func definition GetBuildGuidance/AddBuildGuidance, etc where 2.0.0 and 2.1.0 will have their own implementation of the func.

This is exactly like the outermost `DevfileData` but one level nested to handle different implementation versions between 2.0.0 & 2.1.0. Check the New func in pkg/devfile/parser/data/v2/helper.go where we return either of the above structs to the new empty interface(which will eventually have funcs for new features in 2.1.0)

Reply:
You may check this commit db34d9f4ecc1802426e7ae1962fbeb0f69e6f40f for a better idea of what the above paragraph just described